image/verify/src/verifier.rs

```diff
@@ -520,10 +522,11 @@ impl<Env: ImageVerificationEnv> ImageVerifier<Env> {
                             self.env.set_fw_extended_error(err.into());
                             CaliptraError::IMAGE_VERIFIER_ERR_VENDOR_LMS_VERIFY_FAILURE
                         })?;
-                if cfi_launder(candidate_key) != HashValue::from(lms_pub_key.digest) {
+                let pub_key_digest = HashValue::from(lms_pub_key.digest);
+                if candidate_key != pub_key_digest {
```

Comment:
candidate_key should be laundered. Most likely the else branch is optimized out.

Reply:
It can't optimize out the else branch; it's calling assembly code which is opaque to the compiler.

cfi_launder shouldn't make a difference here; the compiler still needs to do this comparison (it wasn't done anywhere else in this function).